src/ctap/storage.rs

```diff
@@ -264,12 +265,10 @@ impl PersistentStore {
         debug_assert_eq!(entry.tag, TAG_CREDENTIAL);
         let result = deserialize_credential(entry.data);
         debug_assert!(result.is_some());
-        if check_cred_protect
-            && result.as_ref().map_or(false, |cred| {
-                cred.cred_protect_policy
-                    == Some(CredentialProtectionPolicy::UserVerificationRequired)
-            })
-        {
+        let user_verification_required = result.as_ref().map_or(false, |cred| {
+            cred.cred_protect_policy == Some(CredentialProtectionPolicy::UserVerificationRequired)
+        });
+        if check_cred_protect && user_verification_required {
```

Comment:
Might not use short-circuiting this way? (I don't particularly mind.)

Reply:
With the 1-statement if with `&&`, Clippy complains about the complexity. If I nest two ifs, it complains that it can be collapsed into a `&&`. Here we don't really care about the short-circuiting so I think naming the result of this `map_or` as `user_verification_required` and then using it is the clearest.